lib/twirp/encoding.rb

```diff
@@ -32,6 +37,7 @@ def decode(bytes, msg_class, content_type)
       def encode(msg_obj, msg_class, content_type)
         case content_type
         when JSON  then msg_class.encode_json(msg_obj, emit_defaults: true)
+        when JSON_STRICT then msg_class.encode_json(msg_obj, emit_defaults: true)
```

Comment:
`when JSON, JSON_STRICT then ...` maybe better in the same statement because it is the same?
Granted, it seems that with this option, it would make more sense to use `emit_defaults: true` only when JSON_STRICT. What do you think?

Reply:
Interesting -- I'm a fan of this though my only concern is to be sure it doesn't make a behavior change.  If someone is using JSON mode and we stop emitting defaults then the way ruby parses protos some messages would be nil that were once filled in.

To be fair that's probably not the end of the world but I could see it being somewhat unexpected.  Thoughts?

Reply:
I've thought a bit more on this, this is definitely something that can happen but I don't think there is a huge risk in getting rid of emit_defaults in JSON mode.  I will note it'll be super annoying though to hit this bug if you don't have context.  Going to make the change but happy to revert it if you change your mind.

Reply:
Okay I thought about this even more, feels sketchy and confusing.  I'd actually really like to get rid of emit_defaults since it's probably costing bytes for people but now if you test via curl and then call you may get confused.  Also if you're passing this JSON payload to a website you may have depended on the behavior.

I'm happy to change it since I think it's for the best to get rid of it but do you think it requires a major version bump?

Reply:
Updated but want your thoughts per above @marioizquierdo

Reply:
It works either way. Twirp actually didn't have emit_defaults until a few months back when v7 was released (neither on Ruby or Go implementations). That was indeed a common point of confusion when debugging, and that's why we changed it to `emit_defaults: true`. The change is mostly backwards compatible (we didn't see any report of issues because of adding those keys back). It will be fine to change it back to false.

So in one hand it makes sense to keep it as `true` to be closer to the current Go implementation. But in other hand it is fine to use `false` by default because that's how it used to behave before, and now we have a clear option with `strict: true`. So, really up to you, it is fine with me either way.

Reply:
arg, ok, I made a last-minute decision and will change emit_defaults back to true in the regular mode. I see your struggle now 😄  Even tho this is minor thing it is tricky heh. I will merge this and then change it in master in a post commit before making the release.

The rationale is that we should stay closer to the Go implementation by default. We should propose using `strict=true` in Go, and then if that is accepted and the Go implementation decides to change emit_defaults back to false then we can do that in here as well.